include/nuttx/queue.h

```diff
@@ -107,6 +107,17 @@
     } \
   while (0)
 
+#define dq_addfirst_notempty(p, q) \
```

Comment:
```suggestion
#define dq_addfirst_nonempty(p, q) \
```
move to sched/

Reply:
let's add sched/sched/queue.h and move the new special macro to that file.